### DISCUSSION TITLE
Plugin development and testing

### DISCUSSION THREAD

**Discussion by horuskol** [tailwindlabs/tailwindcss#8166] [Ideas]
I was working on enabling dynamic utilities / arbitrary values into a plugin, and I'm taking a test-driven approach.

I worked from some of the tests in the Tailwind CSS core to see how to handle arbitrary values, and have managed to get tests working. However, the core tests use helper code that isn't included in the npm package: including custom CSS matchers for Jest, and also easy ways to configure and run Tailwind CSS in the test.

I wonder if it would be possible to create a small package which provides these test helpers to assist with plugin development?

**Comment by cossssmin** [comment]
I'm 2 years late but just found your post, so in case others find it useful I made a plugin starter project for Tailwind CSS:

https://github.com/cossssmin/tailwindcss-plugin-starter

It uses Vitest which has a Jest-compatible API, and it also includes a custom CSS matcher.

Hope this helps! :)
